Add `OTEL_SERVICE_NAME` environment variable (#1677)
open-telemetry/opentelemetry-specification: specification/resource/semantic_conventions/README.md

Certain attribute groups in this document have a **Required** column. For these groups if any attribute from the particular group is present in the Resource then all attributes that are marked as Required MUST be also present in the Resource. However it is also valid if the entire attribute group is omitted (i.e. none of the attributes from the particular group are present even though some of them are marked as Required in this document).

## Attributes with Special Handling

> Given their significance some resource attributes receive a special treatment as described below.

Reviewer: Nit: This sounds quite colloquial (at least for me as a non-native speaker). How about this?
```suggestion
Given their significance some resource attributes are treated specifically as described below.
```
